design/secrets_provider_milestone1_2_design.md

```diff
@@ -0,0 +1,380 @@
+# Secrets Provider - Phase 2
+
+## Table of Contents
+- [Useful links](#useful-links)
+- [Background](#background)
+  * [Motivation](#motivation)
+  * [Requirement](#requirement)
+- [Solution](#solution)
+  * [Milestone 1: Deployment](#milestone-1--deployment)
+    + [Design](#design)
+    + [Flow](#flow)
+    + [Customer Experience](#customer-experience)
+    + [Code changes](#code-changes)
+    + [Fetching Relevant K8s Secrets](#fetching-relevant-k8s-secrets)
+      - [**Customer Experience**](#--customer-experience--)
+  * [Milestone 2: Update K8s secret with Conjur secret](#milestone-2--update-k8s-secret-with-conjur-secret)
+    + [Design](#design-1)
+    + [Flow](#flow-1)
+    + [Customer experience](#customer-experience)
+    + [Code changes](#code-changes-1)
+  * [Order of Deployment](#order-of-deployment)
+      - [Milestone 1 and 2](#milestone-1-and-2)
+  * [Backwards compatibility](#backwards-compatibility)
+      - [Milestone 1 and 2](#milestone-1-and-2-1)
+  * [Performance](#performance)
+  * [Affected Components](#affected-components)
+      - [Milestone 1](#milestone-1)
+      - [Milestone 2](#milestone-2)
+- [Security](#security)
+- [Test Plan](#test-plan)
+- [Logs](#logs)
+  * [Audit](#audit)
+- [Documentation](#documentation)
+- [Open questions](#open-questions)
+- [Implementation plan](#implementation-plan)
+
+
+
+| Name                | Link                                                         |
+| ------------------- | ------------------------------------------------------------ |
+| Phase 2 Feature Doc | https://github.com/cyberark/secrets-provider-for-k8s/issues/102 |
+
+## Background
+The Cyberark Secrets Provider for K8s enables customers to use secrets stored and managed in Conjur/DAP vault and consume them as K8s Secrets for their application containers. During Phase 1, an *init* container was deployed in the same Pod as the customer's applications and populate the K8s secrets defined in the customer's manifest. 
+
+### Motivation
+
+The current implementation makes the deployment of the Secrets Provider does not support secret rotation and makes the deployment of the Secrets Provider deeply coupled to the customer's applications which not all customers want.
+
+### Requirement
+
+The solution should support the following capabilities:
+
+1. Deployment, deploy as separate Conjur entity; serve K8s Secrets to many Applications
```

Comment:
in the same namespace or do we now serve more than one namespace?

Reply:
Only current namespace